diff -Naur --exclude '*~' mozart-1.4.0.20080704/platform/emulator/urlc.cc mozart-1.4.0.20080704-fix-build/platform/emulator/urlc.cc
--- mozart-1.4.0.20080704/platform/emulator/urlc.cc	2005-05-19 02:23:12.000000000 +0200
+++ mozart-1.4.0.20080704-fix-build/platform/emulator/urlc.cc	2009-06-10 16:01:58.000000000 +0200
@@ -179,10 +179,10 @@
     int parse_file(const char* line);
     int get_file(void);
 
-    int parse_http(const char* line);
+    int parse_http(char* line);
     int get_http(char *file);
 
-    int parse_ftp(const char* line);
+    int parse_ftp(char* line);
     int get_ftp(char *file);
 
     int parse(const char* line);
@@ -595,7 +595,7 @@
    be % escaped _before_ given to this client
    */
 int
-urlc::parse_ftp(const char* line)
+urlc::parse_ftp(char* line)
 {
     char* p_collon = NULL; /* positions in line */
     char* p_at = NULL;
@@ -1124,7 +1124,7 @@
    returns URLC_OK on successful parse, reason otherwise.
    */
 int
-urlc::parse_http(const char* line)
+urlc::parse_http(char* line)
 {
     char* p_collon = NULL; /* positions in line */
     char* p_slash = NULL;
